File: make-zip.py
import zipfile
import os

output = 'cloudbase-api-v3.zip'

with zipfile.ZipFile(output, 'w', zipfile.ZIP_DEFLATED) as zf:
    # 添加单个文件
    zf.write('cloudbase-api/index.js', 'index.js')
    zf.write('cloudbase-api/package.json', 'package.json')
    zf.write('cloudbase-api/scf_bootstrap', 'scf_bootstrap')
    
    # 递归添加 node_modules
    for root, dirs, files in os.walk('cloudbase-api/node_modules'):
        for file in files:
            full_path = os.path.join(root, file)
            arcname = os.path.relpath(full_path, 'cloudbase-api').replace('\\', '/')
            zf.write(full_path, arcname)

print(f'Zip created: {output}')

File: docs/make-zip.js
const fs = require('fs');
const path = require('path');
const archiver = require('archiver');

const output = fs.createWriteStream('cloudbase-api-v3.zip');
const archive = new archiver.Archiver('zip', { zlib: { level: 9 } });

output.on('close', () => {
  console.log('Zip created: ' + archive.pointer() + ' bytes');
});

archive.on('error', (err) => {
  throw err;
});

archive.pipe(output);

// 添加单个文件
archive.file('cloudbase-api/index.js', { name: 'index.js' });
archive.file('cloudbase-api/package.json', { name: 'package.json' });
archive.file('cloudbase-api/scf_bootstrap', { name: 'scf_bootstrap' });

// 递归添加 node_modules
function addDir(srcPath, zipPath) {
  const files = fs.readdirSync(srcPath);
  for (const file of files) {
    const fullPath = path.join(srcPath, file);
    const stat = fs.statSync(fullPath);
    const zipName = zipPath ? zipPath + '/' + file : file;
    if (stat.isDirectory()) {
      addDir(fullPath, zipName);
    } else {
      archive.file(fullPath, { name: zipName });
    }
  }
}

addDir('cloudbase-api/node_modules', 'node_modules');

archive.finalize();
